### ISSUE TITLE
Rule should not trigger on enums

### ISSUE THREAD

**Issue by tylerlaprade** [ArnaudBarre/eslint-plugin-react-refresh#81]
When I make changes in my file containing an enum, I don't see any warnings about HMR in Vite. This rule should therefore not trigger on enums when `allowConstantExport` is enabled.

**Comment by ArnaudBarre** [comment]
Can you provide a small example? If you mean TS enum this is probably more a bug in the Vite HMR

**Comment by tylerlaprade** [comment]
Sure, here is an example. I have this enum in the same file as a React component.
```tsx`
export enum RecordStyle {
  STACKED,
  HORIZONTAL,
}
```
I'm curious what you mean by "a bug in the Vite HMR". Given that I do _not_ have any errors when HMR'ing changes to this file, it seems like Vite is handling this just fine, the same as it does for constant literal exports.

**Comment by ArnaudBarre** [comment]
Do you have an example for that: when I add this enum to the starter template I correctly get a hmr warning on update: https://stackblitz.com/edit/vitejs-vite-5tui148d?file=src%2FApp.tsx&terminal=dev

**Comment by ArnaudBarre** [comment]
I'm closing because this is working as expected for me. If you find a pattern that don't make enum report an HMR issue, you can open an issue in https://github.com/vitejs/vite-plugin-react/issues because this should not currently be allows

**Comment by tylerlaprade** [comment]
Aha, I am indeed seeing those warnings now. I'm not sure whether something was off before or I just missed it. Thanks!

<img width="379" alt="Image" src="https://github.com/user-attachments/assets/07b87092-cec9-43cb-a358-5c4d794310cd" />
